src/platforms/hosted/remote_jtagtap.c

```diff
@@ -156,8 +151,10 @@ static void jtagtap_tdi_tdo_seq(
 		}
 		if (DO) {
 			uint64_t res = remotehston(-1, (char *)&construct[1]);
-			memcpy(DO, &res, bytes);
-			DO += bytes;
+			for (i = bytes; i > 0; i--) {
```

Comment:
This results in sub-par code gen - the approach we linked was suggested specifically because you'll get the best possible code generation that way - we've already spent hours making sure that was so.

Reply:
Maybe this is not the optimal code, but it is clear and readable and
 1. We wasted  before a lot of cycles for the USB transaction
 2. This call can not fill up an USB request with more than 8 bytes and is tagged with "FIXME: Provide and test faster call and keep fallback for old firmware"